packages/lit-task-auto-top-up/src/taskHandler.ts

```diff
@@ -0,0 +1,56 @@
+import { Job } from '@hokify/agenda';
+import awaity from 'awaity';
+import consola from 'consola';
+import VError from 'verror';
+
+import fetchRecipientList from './actions/getRecipientList';
+import mintCapacityCreditNFT from './actions/mintCapacityCreditNFT';
+import transferCapacityTokenNFT from './actions/transferCapacityTokenNFT';
+import { toErrorWithMessage } from './errors';
+import { printTaskResultsAndFailures, tryTouchTask } from './taskHelpers';
+import { TaskResult, RecipientDetail } from './types/types';
+
+const { mapLimit } = awaity;
+
+export async function handleRecipient({
+  recipientDetail,
+}: {
+  recipientDetail: RecipientDetail;
+}): Promise<TaskResult> {
+  const { recipientAddress } = recipientDetail;
+
+  const capacityTokenIdStr = await mintCapacityCreditNFT({ recipientDetail });
+  await transferCapacityTokenNFT({ capacityTokenIdStr, recipientAddress });
+
+  return { capacityTokenIdStr, ...recipientDetail };
+}
+
+export async function handleTask(task: Job) {
+  consola.log('Running auto-top-up task; fetching recipient list');
+
+  try {
+    const recipientList = await fetchRecipientList();
+    consola.log(`Loaded ${recipientList.length} recipient addresses from JSON`);
+
+    const results = await mapLimit<RecipientDetail, PromiseSettledResult<TaskResult>>(
+      recipientList,
+      async (recipientDetail) => {
+        const [settledResult] = await Promise.allSettled([handleRecipient({ recipientDetail })]);
+
+        consola.log(`Finished top-up for ${recipientDetail.recipientAddress}`);
+        tryTouchTask(task).then(() => true); // Fire-and-forget; touching job is not critical
+
+        return settledResult;
+      },
+      3 // Let us mint 3 at a time so this isn't too slow but isn't a spammy deluge
```

Comment:
@Ansonhkg @FedericoAmura I have a suspicion this concurrency might actually fail due to `nonce` being re-used if multiple requests are sent for the same account concurrently. Can you confirm if that's the case?

Reply:
Yes, this will fail - that's why in the current e2e tests, they are all in async rather than parallel. A previous solution involved queueing operations to prevent this issue - by creating a wrapper function that interacts with a locally hosted API server before the test. This server manages a queue in memory. When the API endpoint is called, it doesn't return a success response (true) until it has completed the previous operation.

Reply:
OK, modified it to use `mapSeries` instead.